job-shop-scheduling/job_shop_scheduler.py

```diff
@@ -224,20 +225,31 @@ def _add_share_machine_constraint(self):
     def _remove_absurd_times(self):
         """Sets impossible task times in self.csp to 0.
         """
-        # TODO: deal with overlaps in time
-        for task in self.tasks:
-            # Times that are too early for task
-            for t in range(task.position):
-                label = get_label(task, t)
-                self.csp.fix_variable(label, 0)
-
-            # Times that are too late for task to complete
-            for t in range(task.duration - 1):  # -1 to ignore duration==1
-                label = get_label(task, (self.max_time - 1) - t)  # -1 for zero-indexed time
-                self.csp.fix_variable(label, 0)
-
-    def get_bqm(self):
+        jobs = set([task.job for task in self.tasks])
+        for job in jobs:
+            job_tasks = sorted([task for task in self.tasks if task.job == job], key=lambda x: x.position)
+            predecessor_time = 0
+            successor_time = sum([job_task.duration for job_task in job_tasks])
+            for job_task in job_tasks:
+                # Times that are too early for task
+                for t in range(predecessor_time):
+                    label = get_label(job_task, t)
+                    self.csp.fix_variable(label, 0)
+
+                # Times that are too late for task
+                for t in range(successor_time - 1): # -1 to ignore duration==1
+                    label = get_label(job_task, (self.max_time - 1) - t) # -1 for zero-indexed time
+                    self.csp.fix_variable(label, 0)
+
+                # Adjust times
+                predecessor_time += job_task.duration
+                successor_time -= job_task.duration
+
+    def get_bqm(self, stitch_kwargs={}):
```

Comment:
Default arguments should not be mutable. See [here](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) for details.

Instead, consider using
```
def get_bqm(self, stitch_kwargs=None):
    if stitch_kwargs is None:
        stitch_kwargs = {}
    ..
```

Reply:
FYI, I'm currently still reviewing this PR. I just clicked too quickly (i.e. clicked "clicked add single comment" rather than "start a review"). More suggestions to come.

Also, thank you for helping to improve this demo!

Reply:
Thanks for you comments on the changes so far!

I've addressed most of issues so far, but I have a couple where I had a few extra comments/questions. Once we figure those out I'll make the rest of the changes.